ecl/hql/hqlgram2.cpp

```diff
@@ -12341,41 +12346,54 @@ void parseAttribute(IHqlScope * scope, IFileContents * contents, HqlLookupContex
     if (parsed && !alreadySimplified)
     {
         //Forward scopes cannot be cached because the dependencies are not known as it is parsed
-        const bool canCache = parsed->getOperator() != no_forwardscope;
+        const bool canCache = parsed->getOperator() != no_forwardscope ;
         if (canCache)
         {
             const bool isMacro = parsed->isMacro();
-            bool updateCache = ctx.hasCacheLocation() && (!cacheUptoDate || ctx.regenerateCache());
-            bool useSimplified = ctx.syntaxChecking() && !ctx.ignoreSimplified();
+            const bool updateCache = ctx.hasCacheLocation() && (!cacheUptoDate || ctx.regenerateCache());
+            const bool simplifiedRequired = ctx.syntaxChecking() && !ctx.ignoreSimplified() && !useUnsimplified;
 
             OwnedHqlExpr simplified;
             StringBuffer simplifiedEcl;
             if (!isMacro && !ctx.ignoreSimplified() &&
-                (updateCache || ctx.checkSimpleDef() || useSimplified) && !ctx.neverSimplify(fullName))
+                (updateCache || ctx.checkSimpleDef() || simplifiedRequired) && !ctx.neverSimplify(fullName))
+            {
                 simplified.setown(createSimplifiedDefinition(parsed));
+                useUnsimplified = false;
+            }
+            else
+                useUnsimplified = true;
```

Comment:
Note:  UseUnsimplified flag is written out to the cache file.  This is normally okay.  However, if compiler is started with neverSimply= and the recompiled without the neverSimplify option, it will continue use the unsimplified definition because it will have picked up the UseUnsimplified flag from the cache file (line 12311).  We'll need to discuss if this behaviour is acceptable.